src/EpiEasyEvents/EventsRegistry.cs

```diff
@@ -3,63 +3,81 @@
 using System.Linq;
 using EPiServer;
 using EPiServer.Core;
+using EPiServer.DataAbstraction;
 using Forte.EpiEasyEvents.EventHandlers;
 using Microsoft.Extensions.DependencyInjection;
 
 namespace Forte.EpiEasyEvents
 {
     internal class EventsRegistry
     {
-        private readonly IContentEvents _contentEvents;
         private readonly IServiceProvider _services;
+        private readonly IContentLoader _contentLoader;
+        private readonly IContentEvents _contentEvents;
+        private readonly IContentSecurityEvents _contentSecurityEvents;
 
-        public EventsRegistry(IContentEvents contentEvents, IServiceProvider services)
+        public EventsRegistry(IServiceProvider services, IContentLoader contentLoader, IContentEvents contentEvents, IContentSecurityEvents contentSecurityEvents)
```

Comment:
I wonder, do we need to inject `IContentLoader` since we already have all available services in `IServiceProvider` and we can get `IContentLoader` as `services.GetService<IContentLoader>()`?

Reply:
I do prefer injecting it through the constructor as it is way easier to mock it in the unit tests. In general, it could cause some issues if we inject non-singleton dependency here as the EventsRegistry is registered as a singleton - but ContentLoader is actually a singleton as well, so it is not the case.